apps/sms/views/conversation/test/unit/conversation_test.js

```diff
@@ -6786,6 +6783,8 @@ suite('conversation.js >', function() {
         ConversationView.initRecipients();
         Navigation.isCurrentPanel.withArgs('composer').returns(true);
         this.sinon.stub(ConversationView.recipients, 'focus');
+        this.sinon.stub(Compose, 'focus');
+        this.sinon.stub(ConversationView, 'hasValidRecipients');
```

Comment:
In the term "unit testing", we're testing a "unit of code"; the unit of code is the whole object ConversationView.
So ideally we should not mock other methods of this object while testing a method of this object. We don't need to test how the unit works internally, this is not what's important. What's important is how the unit works with inputs/outputs.

Also this makes it easier to refactor the unit.

if we feel like we need to, it usually means the object holds several responsiblities and we should split it.

In this case, I don't think it's the case though. It's just as easy to add something (or not) to ConversationView.recipients.numbers (I think ConversationView.recipients is a MockRecipients, but please check this).

Hope this is clearer !